core/src/main/java/cucumber/runtime/order/RandomOrderType.java

```diff
@@ -0,0 +1,39 @@
+package cucumber.runtime.order;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import cucumber.runtime.CucumberException;
+import gherkin.events.PickleEvent;
+
+public class RandomOrderType extends OrderType {

```

Comment:
This should be an Enum that implements/extends `OrderType`.

Reply:
https://stackoverflow.com/questions/2709593/why-would-an-enum-implement-an-interface

Reply:
Will convert into enum.  No need of Factory.